services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,250 @@
+            it.portNumber == isl.srcPort
+            it.discoveryEnabled == newDisoveryEnabled
+        }
+
+        and: "Discovery port property is really updated"
+        with(northboundV2.getPortProperties(isl.srcSwitch.dpId, isl.srcPort)) {
+            it.switchId == isl.srcSwitch.dpId
+            it.portNumber == isl.srcPort
+            it.discoveryEnabled == newDisoveryEnabled
+        }
+
+        cleanup: "Restore init discovery port property on the port"
+        northboundV2.updatePortProperties(isl.srcSwitch.dpId, isl.srcPort,
+                new PortPropertiesDto(discoveryEnabled: initPortProperties.discoveryEnabled))
+    }
+
+    def "Informative error is returned when trying to get/update port properties with non-existing switch"() {
+        when: "Try to get port properties info for non-existing switch"
+        //assume port 10 is always exist on a switch
+        def port = 10
+        northboundV2.getPortProperties(NON_EXISTENT_SWITCH_ID, port)
+
+        then: "Human readable error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage ==
+                "Port properties not found: No switch found with id '${NON_EXISTENT_SWITCH_ID}'"
+
+        when: "Try to update discovery port property info for non-existing switch"
+        northboundV2.updatePortProperties(NON_EXISTENT_SWITCH_ID, port, new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update port properties for '${NON_EXISTENT_SWITCH_ID}_${port}':\
+ No switch found with id '${NON_EXISTENT_SWITCH_ID}'"
+    }
+
+    def "Informative error is returned when trying to update port properties with non-existing port number"() {
+        when: "Try to get port properties info for non-existing port"
+        def sw = topology.activeSwitches.first()
+        def nonExistentPort = 99999
+        def response = northboundV2.getPortProperties(sw.dpId, nonExistentPort)
+
+        then: "No error, default port propreties information is returned"
+        with(response) {
+            it.switchId == sw.dpId
+            it.portNumber == nonExistentPort
+            it.discoveryEnabled
+        }
+
+        when: "Try to update discovery port property info for non-existing port"
+        northboundV2.updatePortProperties(sw.dpId, nonExistentPort, new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Port not found: 'Port FSM not found (${sw.dpId}_${nonExistentPort}).'"
+    }
```

Comment:
Yes, It's a bit strange from business perspective but from tech perspective port validation for get request would require refactoring

Reply:
Any plans for future updates of this behavior? Anyway any strange behavior requires a comment in the test. With or without a link to future plans of improving this I'd like to see an explicit explanation that this is exactly how it is intended to work right now (since the behavior is weird, not obvious)

Reply:
comment was added at the beginning of the test